mysensors/mysensors.py

```diff
@@ -49,6 +50,22 @@ def __init__(self, event_callback=None, persistence=False,
         if persistence:
             self._safe_load_sensors()
 
+    @property
+    def event_callback(self):
+        """Getter for event_callback property."""
+        return self._event_callback
+
+    @event_callback.setter
+    def event_callback(self, callback):
+        """Setter for event_callback property."""
+        self._event_callback = callback
+        if callback is not None:
+            if len(inspect.signature(callback).parameters) == 2:
+                def adapter(message):
+                    """Event callback adapter for backwards compatibility."""
+                    callback('sensor_update', message.node_id)
+                self._event_callback = adapter
```

Comment:
I agree that a breaking change would result in the cleanest code. However, these changes do still support the existing callbacks (see the related the unit tests) unless there's a case I didn't consider (very possible). The adaptation functionality is hidden from the user. I understand though that there may be possible confusion about the gateway supporting two different event callback signatures. Let me know if you still prefer the breaking change and I'll update the code and related unit tests. Thanks.